carta/image.py

```diff
@@ -426,23 +426,103 @@ def set_polarization(self, polarization, recursive=True):
 
         self.call_action("setChannels", self.macro("", "requiredChannel"), polarization, recursive)
 
-    @validate(Number(), Number())
-    def set_center(self, x, y):
-        """Set the center position.
+    @property
+    @cached
+    def valid_wcs(self):
+        """Whether the image contains valid WCS information.
+
+        Returns
+        -------
+        boolean
+            Whether the image has WCS information.
+        """
+        return self.get_value("validWcs")
+
+    @validate(Coordinate(), Coordinate(), NoneOr(Constant(CoordinateSystem)))
+    def set_center(self, x, y, system=None):
+        """Set the center position, in image or world coordinates. Optionally change the session-wide coordinate system.
 
-        TODO: what are the units?
+        Coordinates must either both be image coordinates or match the current number formats. Numbers and numeric strings with no units are interpreted as degrees.
 
         Parameters
         ----------
         x : {0}
             The X position.
         y : {1}
             The Y position.
+        system : {2}
+            The coordinate system. If this parameter is provided, the coordinate system will be changed session-wide before the X and Y coordinates are parsed.
+
+        Raises
+        ------
+        ValueError
+            If image coordinates are outside the bounds of the image, if a mix of image and world coordinates is provided, or if world coordinates do not match the session-wide number format.
+        """
+        if system is not None:
+            self.session.set_coordinate_system(system)
+
+        x_is_pixel = PixelValue.valid(str(x))
+        y_is_pixel = PixelValue.valid(str(y))
+
+        if x_is_pixel and y_is_pixel:
+            # Image coordinates
+            x_value = PixelValue.as_float(str(x))
+            y_value = PixelValue.as_float(str(y))
+            if 0 <= x_value < self.width and 0 <= y_value < self.height:
```

Comment:
This is minor but I just realized, in CARTA we refer the _center_ of the pixel at the bottom-left corner as (0, 0). This means, for example in x direction we have 100 pixels, the _valid_ range should be -0.5 <= x < 99.5 (or -0.5 <= x <= 99.5?).

Reply:
I decided to make the range symmetrical -- if the user can center the bottom-left corner, they should also be able to center the other corners.

Does it make sense to apply a range limit at all? The user can specify WCS coordinates which are outside the image area, right? Should they not be able to do the same here?

Reply:
Right, world coordinate can be outside the image. So guess we can remove the range check for consistency and potentially enable some use cases.